components/ws-proxy/pkg/proxy/infoprovider.go

```diff
@@ -400,7 +402,24 @@ func (c *workspaceInfoCache) Insert(info *WorkspaceInfo) {
 }
 
 func (c *workspaceInfoCache) doInsert(info *WorkspaceInfo) {
-	c.infos[info.WorkspaceID] = info
+	existingInfos := c.infos[info.WorkspaceID]
```

Comment:
if `c.infos` were a `map[string]map[string]*WorkspaceInfo` the code below could be simplified to
```Go
existingInfos, ok := c.infos[info.WorkspaceID]
if !ok {
    existingInfos = make(map[string]*WorkspaceInfo)
}
existingInfos[info.InstanceID] = info
c.infos[info.WorkspaceID] = existingInfos
```